File: web/src/main/java/hu/bsstudio/robonaut/scores/SeniorEndResultHandler.java
package hu.bsstudio.robonaut.scores;

import hu.bsstudio.robonaut.scores.endresult.EndResultService;
import hu.bsstudio.robonaut.scores.endresult.model.EndResultedTeam;
import hu.bsstudio.robonaut.team.model.DetailedTeam;
import lombok.NonNull;
import lombok.RequiredArgsConstructor;
import org.springframework.web.reactive.function.server.HandlerFunction;
import org.springframework.web.reactive.function.server.ServerRequest;
import org.springframework.web.reactive.function.server.ServerResponse;
import reactor.core.publisher.Mono;

@RequiredArgsConstructor
public class SeniorEndResultHandler implements HandlerFunction<ServerResponse> {

    @NonNull
    private final EndResultService service;

    @Override
    public Mono<ServerResponse> handle(final ServerRequest request) {
        final var detailedTeam = request.bodyToFlux(EndResultedTeam.class)
            .flatMap(service::updateEndResultSenior);
        return ServerResponse.ok().body(detailedTeam, DetailedTeam.class);
    }
}

File: web/src/main/kotlin/hu/bsstudio/robonaut/scores/SeniorEndResultHandler.kt
package hu.bsstudio.robonaut.scores

import hu.bsstudio.robonaut.scores.endresult.EndResultService
import hu.bsstudio.robonaut.scores.endresult.model.EndResultedTeam
import org.springframework.web.reactive.function.server.*
import reactor.core.publisher.Mono

class SeniorEndResultHandler(private val service: EndResultService) : HandlerFunction<ServerResponse> {

    override fun handle(request: ServerRequest): Mono<ServerResponse> {
        return request.bodyToFlux<EndResultedTeam>()
            .flatMap(service::updateEndResultSenior)
            .let { detailedTeam -> ServerResponse.ok().body(detailedTeam) }
    }
}
